lib/bot.js

```diff
@@ -98,6 +100,24 @@ class Bot {
       this.sendToDiscord(author, to, `*${text}*`);
     });
 
+    this.ircClient.on('join', (channel, nick) => {
+      if (this.disableJoinPartMsgs) return;
+      if (nick === this.nickname && !this.announceSelfJoin) return;
```

Comment:
Didn't know you'd receive join events for yourself. Do we need a configuration option for this though? There's already quite a lot of configuration possible, so maybe it would be better to just keep this off.